docs/pages/guides/FormikOurValidation.example.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import Code from '../../_shared/Code';
+import { Grid } from '@material-ui/core';
+import { TextField } from '@material-ui/core';
+import { Formik, Form, Field, FieldProps } from 'formik';
+import { format, isWeekend, isWednesday } from 'date-fns';
+import { DatePicker, DatePickerProps } from '@material-ui/pickers';
+
+interface DatePickerFieldProps extends FieldProps, DatePickerProps {
+  getShouldDisableDateError: (date: Date) => string;
+}
+
+const DatePickerField: React.FC<DatePickerFieldProps> = ({
+  form,
+  field: { value, name },
+  maxDate = new Date('2100-01-01'),
+  minDate = new Date('1900-01-01'),
+  getShouldDisableDateError,
+  ...other
+}) => {
```

Comment:
For consistency

```suggestion
function DatePickerField(props: DatePickerFieldProps) {
  const {
    form,
    field: { value, name },
    maxDate = new Date('2100-01-01'),
    minDate = new Date('1900-01-01'),
    getShouldDisableDateError,
    ...other
  } = props;
```

Reply:
What do you mean under consistency? `function` vs `const`?

Reply:
I mean that we systematically write new private components following this pattern:

```jsx
function DatePickerField(props: DatePickerFieldProps) {
  const {
    form,
    field: { value, name },
    maxDate = new Date('2100-01-01'),
    minDate = new Date('1900-01-01'),
    getShouldDisableDateError,
    ...other
  } = props;
```

and for public ones:

```jsx
const DatePickerField = React.forwardRef(function DatePickerField(
  props: DatePickerFieldProps,
  ref: React.Ref<HTMLElement>,
) {
  const {
    form,
    field: { value, name },
    maxDate = new Date('2100-01-01'),
    minDate = new Date('1900-01-01'),
    getShouldDisableDateError,
    ...other
  } = props;
```